File: apps/mybookkeeper/frontend/src/shared/types/lease/signed-lease-summary.ts
import type { SignedLeaseStatus } from "@/shared/types/lease/signed-lease-status";

export interface SignedLeaseSummary {
  id: string;
  user_id: string;
  organization_id: string;
  /** Ordered template IDs (display_order ascending). Empty for imported leases. */
  template_ids: string[];
  applicant_id: string;
  listing_id: string | null;
  kind: "generated" | "imported";
  status: SignedLeaseStatus;
  starts_on: string | null;
  ends_on: string | null;
  generated_at: string | null;
  signed_at: string | null;
  created_at: string;
  updated_at: string;

  /** Decrypted applicant name from applicants.legal_name. Null if not set. */
  applicant_legal_name: string | null;
}

File: apps/mybookkeeper/backend/app/schemas/leases/signed_lease_summary.py
"""Minimal signed-lease payload for list views."""
from __future__ import annotations

import datetime as _dt
import uuid

from pydantic import BaseModel, ConfigDict


class SignedLeaseSummary(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    organization_id: uuid.UUID
    template_id: uuid.UUID | None = None
    applicant_id: uuid.UUID
    listing_id: uuid.UUID | None = None
    kind: str
    status: str
    starts_on: _dt.date | None = None
    ends_on: _dt.date | None = None
    generated_at: _dt.datetime | None = None
    signed_at: _dt.datetime | None = None
    created_at: _dt.datetime
    updated_at: _dt.datetime

    # Denormalised from applicants.legal_name for the leases list page.
    # Decrypted at load time by the EncryptedString TypeDecorator.
    # NULL when the applicant has been soft-deleted or has no legal_name set.
    applicant_legal_name: str | None = None

    model_config = ConfigDict(from_attributes=True)
